File: 0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack=new Stack<>();
        int n=s.length();
        for(int i=0;i<n;i++)
        {
            char ch=s.charAt(i);
            if(stack.isEmpty())
            {
                stack.push(ch);
                continue;
            }
            if((ch==')' && stack.peek()=='(') || (ch==']' && stack.peek()=='[') || (ch=='}' && stack.peek()=='{')  )
            {
                stack.pop();
            }
            else{
                stack.push(ch);
            }
        }
        return stack.size()==0;
    }
}

File: 0020-valid-parentheses/solution.c
bool isValid(char* s) {
    int len=strlen(s);
    int stack[len];
    int top=-1;
    for(int i=0;i<len;i++)
    {
        if(s[i]=='(' || s[i]=='[' || s[i]=='{')
        {
            stack[++top]=s[i];
        }
        else if((s[i]==')' && top>=0 && stack[top]=='(') || (s[i]==']' && top>=0 && stack[top]=='[') || (s[i]=='}' && top>=0 && stack[top]=='{'))
        {
            --top;
        }
        else
            return 0;
    }
    return top==-1;
}
